core/storage/blog/gae_models.py

```diff
@@ -0,0 +1,397 @@
+        """
+        return cls.query(
+            cls.author_id == user_id
+        ).get(keys_only=True) is not None
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model is exported as multiple instances per user since there can
+        be multiple blog models relevant to a user.
+        """
+        return base_models.MODEL_ASSOCIATION_TO_USER.MULTIPLE_INSTANCES_PER_USER
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model contains data corresponding to a user to export."""
+        return dict(super(BlogPostModel, cls).get_export_policy(), **{
+            # We do not export the author_id because we should not
+            # export internal user ids.
+            'author_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'title': base_models.EXPORT_POLICY.EXPORTED,
+            'content': base_models.EXPORT_POLICY.EXPORTED,
+            'url_fragment': base_models.EXPORT_POLICY.EXPORTED,
+            'tags': base_models.EXPORT_POLICY.EXPORTED,
+            'thumbnail_filename': base_models.EXPORT_POLICY.EXPORTED,
+            'published_on': base_models.EXPORT_POLICY.EXPORTED,
+        })
+
+    @classmethod
+    def generate_new_blog_id(cls):
+        """Generates a new blog ID which is unique and is in the form of
+        random hash of 12 chars.
+
+        Returns:
+            str. A blog ID that is different from the IDs of all
+            the existing blogs.
+
+        Raises:
+            Exception. There were too many collisions with existing blog IDs
+                when attempting to generate a new blog ID.
+        """
+
+        for _ in python_utils.RANGE(base_models.MAX_RETRIES):
+            blog_id = utils.convert_to_hash(
+                python_utils.UNICODE(
+                    utils.get_random_int(base_models.RAND_RANGE)),
+                base_models.ID_LENGTH)
+            if not cls.get_by_id(blog_id):
+                return blog_id
+        raise Exception(
+            'New blog id generator is producing too many collisions.')
+
+    @classmethod
+    def create(cls, blog_id):
+        """Creates a new BlogDataModel entry.
+
+        Args:
+            blog_id: str. Blog ID of the newly-created blog.
+
+        Returns:
+            BlogDataModel. The newly created BlogDataModel
+            instance.
+
+        Raises:
+            Exception. A blog with the given blog ID exists already.
+        """
+        if cls.get_by_id(blog_id):
+            raise Exception('A blog with the given blog ID exists already.')
+        return cls(id=blog_id)
+
+    # If the user directly enters the blog post's url in the editor or the
+    # homepage, the blogPostModel will be queried using the url fragment to
+    # retrieve data for populating the editor dashboard / blog post page.
+    @classmethod
+    def get_by_url_fragment(cls, url_fragment):
+        """Gets BlogModel by url_fragment. Returns None if the blog with
```

Comment:
```suggestion
        """Gets BlogPostModel by url_fragment. Returns None if the blog with
```

Please take care of naming!

Reply:
Sorry, fixed.